### DISCUSSION TITLE
IfcOpenShell - Visual Studio 2022 solution build error in Debug cofiguration

### DISCUSSION THREAD

**Discussion by SP-AWN** [IfcOpenShell/IfcOpenShell#5765] [Q&A] [answered]
Hi Im trying to build sources from GitHub and Im getting bunch of errors.
What I have done:
set IFCOS_USE_OCCT=TRUE
build-deps.cmd vs2022 Debug
I have modified  cmake/CmakeLists.txt:
set(CMAKE_BUILD_TYPE "Debug")
I have turned off BUILD_IFCPYTHON, BUILD_CONVERT, BUILD_EXAMPLES
then:
run-cmake vs2022-x64 
After that I have launched IfcOpenShell.sln 
Building result:

2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(174,24): error C3878: syntax error: unexpected token 'json_obj' following 'expression'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(174,24): message : error recovery skipped: 'identifier = identifier  . . . ('
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(174,57): error C2760: syntax error: ')' was unexpected here; expected ';'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(174,57): error C3878: syntax error: unexpected token ')' following 'expression-statement'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(174,57): message : error recovery skipped: ')'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(174,35): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(174,9): error C7510: 'json': use of dependent type name must be prefixed with 'typename'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,13): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,28): error C3878: syntax error: unexpected token 'pair' following 'expression'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,28): message : error recovery skipped: 'identifier = identifier  . . . ('
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,57): error C2760: syntax error: ')' was unexpected here; expected ';'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,57): error C3878: syntax error: unexpected token ')' following 'expression-statement'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,57): message : error recovery skipped: ')'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,35): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(180,13): error C7510: 'json': use of dependent type name must be prefixed with 'typename'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,13): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,28): error C3878: syntax error: unexpected token 'coord' following 'expression'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,28): message : error recovery skipped: 'identifier = identifier  . . . ('
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,58): error C2760: syntax error: ')' was unexpected here; expected ';'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,58): error C3878: syntax error: unexpected token ')' following 'expression-statement'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,58): message : error recovery skipped: ')'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,36): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(181,13): error C7510: 'json': use of dependent type name must be prefixed with 'typename'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,13): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,28): error C3878: syntax error: unexpected token 'values' following 'expression'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,28): message : error recovery skipped: 'identifier = identifier  . . . ('
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,59): error C2760: syntax error: ')' was unexpected here; expected ';'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,59): error C3878: syntax error: unexpected token ')' following 'expression-statement'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,59): message : error recovery skipped: ')'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,37): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(182,13): error C7510: 'json': use of dependent type name must be prefixed with 'typename'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,21): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,36): error C3878: syntax error: unexpected token 'coord' following 'expression'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,36): message : error recovery skipped: 'identifier = identifier  . . . ('
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,66): error C2760: syntax error: ')' was unexpected here; expected ';'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,66): error C3878: syntax error: unexpected token ')' following 'expression-statement'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,66): message : error recovery skipped: ')'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,44): error C2065: 'nlohmann': undeclared identifier
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\graph_2d.h(193,21): error C7510: 'json': use of dependent type name must be prefixed with 'typename'
2>(compiling source file '../../src/svgfill/src/arrange_polygons.cpp')
2>D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\arrange_polygons.cpp(255,18): warning C4456: declaration of 'it' hides previous local declaration
2>    D:\DEV_space_AWN\IfcOpenshell-0.8.0\src\svgfill\src\arrange_polygons.cpp(242,15):
2>    see declaration of 'it'
2>Done building project "libsvgfill.vcxproj" -- FAILED.
3>------ Build started: Project: svgfill, Configuration: Debug x64 ------
3>LINK : warning LNK4075: ignoring '/INCREMENTAL' due to '/FORCE' specification
3>LINK : fatal error LNK1181: cannot open input file 'Debug\libsvgfill.lib'
3>D:\C\Visual Studio\2022\Comunity\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(1151,5): error MSB6006: "link.exe" exited with code 1181.
3>Done building project "svgfill.vcxproj" -- FAILED.
4>------ Skipped Build: Project: PACKAGE, Configuration: Debug x64 ------
4>Project not selected to build for this solution configuration 
5>------ Skipped Build: Project: INSTALL, Configuration: Debug x64 ------
5>Project not selected to build for this solution configuration 
========== Build: 0 succeeded, 2 failed, 16 up-to-date, 3 skipped ==========
========== Build completed at 11:47 AM and took 58.164 seconds ==========

**Comment by aothms** [comment, marked as answer]
Thanks, was already repored as #5731 fixed just now.

**Reply by SP-AWN** [reply]
Thanks
